docs/secure-agent-communication.asciidoc

```diff
@@ -122,28 +122,70 @@ privileges.
 
 The user creating an APM agent API key must have at least the `manage_own_api_key` cluster privilege
 and the APM application-level privileges that it wishes to grant.
+In addition, if you're creating an API key from the APM app,
+you'll need the appropriate Kibana Space and Feature privileges.
+
 The example below uses the Kibana {kibana-ref}/role-management-api.html[role management API]
 to create a role named `apm_agent_key_role`.
 
 [source,js]
 ----
 POST /_security/role/apm_agent_key_role
 {
-  "cluster": ["manage_own_api_key"],
-  "applications": [{
-    "application": "apm",
-    "privileges": ["event:write", "config_agent:read"],
-    "resources": ["*"]
-  }]
+   "cluster": [ "manage_own_api_key" ],
+   "applications": [
+      {
+         "application":"apm",
+         "privileges":[
+            "event:write",
+            "config_agent:read"
+         ],
+         "resources":[ "*" ]
+      },
+      { <1>
+         "application":"kibana-.kibana",
+         "privileges":[ "feature_apm.all" ],
+         "resources":[ "space:default" ] <2>
+      }
+   ]
 }
 ----
+<1> The privileges added in this object only apply to roles that will create API keys in the APM app.
+If you don't have access to the APM app, you can omit this object and skip to <<create-an-api-key-alternate>>.
+<2> Assigns privileges for the default space.
 
 Assign the newly created `apm_agent_key_role` role to any user that wishes to create APM agent API keys.
 
 [[create-an-api-key]]
 [float]
-=== Create an API key in {kib}
+=== Create an API key in the APM app
+
+Using a superuser account, or a user with the role created in the previous step,
+open {kib} and navigate to **Observability** > **APM** > **Settings** > **Agent keys**.
+Enter a name for your API key and select at least one privilege.
+
+For example, to create an API key that can be used to ingest APM events
+and read agent central configuration, select `config_agent:read` and `event:write`.
+
+IMPORTANT: The `sourcemap:write` privilege is outdated and will be removed in a future release.
+To learn more about the privileges required to upload a source map,
+see the {kibana-ref}/rum-sourcemap-api.html[RUM source map API].
```

Comment:
I'm assuming we're going to open an issue to remove this privilege from the APM app. In the meantime, I've added this disclaimer.